app/controllers/application_controller.rb

```diff
@@ -34,6 +34,31 @@ def default_html_head
     stylesheet_links << ['argo']
   end
 
+  rescue_from Dor::Services::Client::UnexpectedResponse do |exception|
+    detail = exception.errors.first['detail']&.truncate(1024) # Truncate to avoid cookie overflow. Cookies can hold 4k.
+    message = "Unable to retrieve the cocina model: #{detail}"
+    Honeybadger.notify(exception)
+    logger.error "Error connecting to DSA: #{detail}"
+    if turbo_frame_request?
+      render 'error', locals: { message: }
+    else
+      redirect_to solr_document_path(params[:item_id] || params[:id]),
+                  flash: { error: message }
+    end
+  end
+
+  rescue_from Cocina::Models::ValidationError do |exception|
+    message = "Error building Cocina: #{exception.message.truncate(200)}"
+    Honeybadger.notify(exception)
+    logger.error(message)
+    if turbo_frame_request?
+      render 'error', locals: { message: }
+    else
+      redirect_to solr_document_path(params[:item_id] || params[:id]),
+                  flash: { error: message }
+    end
+  end
```

Comment:
This all shows as uncovered. Worth adding coverage to a request spec? (I know this application controller code applies to all controllers equally.)